src/coredata.cs

```diff
@@ -126,58 +147,74 @@ public interface NSAttributeDescription {
 	public interface NSEntityDescription : NSCoding, NSCopying {
 
 		[Static, Export ("entityForName:inManagedObjectContext:")]
+		[return: NullAllowed]
 		NSEntityDescription EntityForName (string entityName, NSManagedObjectContext context);
 
 		[Static, Export ("insertNewObjectForEntityForName:inManagedObjectContext:")]
-		NSObject InsertNewObjectForEntityForName (string entityName, NSManagedObjectContext context);
+		NSManagedObject InsertNewObjectForEntityForName (string entityName, NSManagedObjectContext context);
```

Comment:
That's a breaking change. 

We can keep the old signature to return a casted `NSManagedObject` into a `NSObject` (inside a `!XAMCORE_4_0` block and with a `[Obsolete]`). 

The new, correct, API could simply be named `InsertNewObject` (the rest can be inferred by the parameters)